### PR TITLE
add functionality to close out dirty PRs and re-issue them

### PR DESCRIPTION
**Pull request by CJ-Wright**
Closes: https://github.com/regro/cf-scripts/issues/1014

### CODE REVIEW

File: conda_forge_tick/git_utils.py [hunk 1]
```diff
@@ -344,3 +346,46 @@ def is_github_api_limit_reached(e: github3.GitHubError, gh: github3.GitHub) -> b
...
+                    ),
+                )
+                pr_obj.close()
+                delete_branch(ctx=ctx, pr_json=pr_json, dry_run=dry_run)
+                pr_obj.refresh(True)
+        d = pr_obj.as_dict()
+        # This will cause the update_nodes_with_bot_rerun to trigger properly and shouldn't be overridden since
+        # this is the last function to run, the long term solution here is to add the bot to conda-forge and then
+        # it should have label adding capability and we can just add the label properly
+        d["labels"].append("bot-rerun")
```
Comment (beckermr) [inline conda_forge_tick/git_utils.py:389]: how does this make it back to disk?

Comment (CJ-Wright) [reply conda_forge_tick/git_utils.py:389]: Thank you!

Comment (CJ-Wright) [reply conda_forge_tick/git_utils.py:389]: We return the dict, which is then used to update the pr information

Comment (beckermr) [inline conda_forge_tick/git_utils.py:389]: the github api has a schema here: we need to emulate it

**Comment by CJ-Wright** [comment]
What do you mean by edits? The bot only closes PRs where it (and the other CF bots) are the sole committers.

**Comment by beckermr** [comment]
NVM - this comment is wrong.

LGTM!

**Pull request merged**